src/edges_cal/reflection_coefficient.py

```diff
@@ -31,162 +50,458 @@ def gamma2impedance(
 ) -> float | np.ndarray:
     """Convert reflection coeffiency to impedance.
 
+    See Eq. 19 of Monsalve et al. 2016.
+
     Parameters
     ----------
-    gamma : float or array
+    gamma
         Reflection coefficient.
-    z0 : float or array
-        Matching impedance (?)
+    z0
+        Impedance of the match.
 
     Returns
     -------
-    z : float or array
+    z
         The impedance.
     """
     return z0 * (1 + gamma) / (1 - gamma)
 
 
-def gamma_de_embed(s11, s12s21, s22, rp):  # noqa
-    return (rp - s11) / (s22 * (rp - s11) + s12s21)
+def gamma_de_embed(
+    s11: np.typing.ArrayLike,
+    s12s21: np.typing.ArrayLike,
+    s22: np.typing.ArrayLike,
+    gamma_ref: np.typing.ArrayLike,
+) -> np.typing.ArrayLike:
+    """Obtain the intrinsic reflection coefficient.
+
+    See Eq. 2 of Monsalve et al., 2016.
+
+    Obtains the instrinsic reflection coefficient from the
+    one measured at the reference plane, given a set of
+    reflection coefficients.
 
+    Parameters
+    ----------
+    s11
+        The reflection coefficient of the two-port
+        network.
+    s12s21
+        The product of ``S12*S21`` of the two-port
+        network.
+    s22
+        The S22 of the two-port network.
```

Comment:
change this description to:
The s22 parameter of the two-port network for the port facing the device under test (DUT).